pkg/dockerregistry/server/manifestservice.go

```diff
@@ -91,6 +91,17 @@ func (m *manifestService) Get(ctx context.Context, dgst digest.Digest, options .
 
 	manifest, err = m.repo.manifestFromImageWithCachedLayers(image, ref.Exact())
 
+	if _, err := m.manifests.Put(WithRepository(ctx, m.repo), manifest, options...); err != nil {
+		context.GetLogger(m.ctx).Errorf("unable to put manifest to storage: %v", err)
+	} else if len(image.DockerImageManifest) > 0 {
+		image.DockerImageManifest = ""
+		image.DockerImageConfig = ""
+
+		if _, err := m.repo.updateImage(image); err != nil {
```

Comment:
I believe the [image strategy](https://github.com/openshift/origin/blob/master/pkg/image/registry/image/strategy.go#L88) needs to be modified to allow manifest removal.